modules/transformation.py

```diff
@@ -0,0 +1,609 @@
+
+        # STEP 2: Create all the resources & associate them with the pageid.
+        entries, first_url, first_html_url = ImportHarJson.import_entries(
+            log["entries"], page_id, status_info
+        )
+        if not entries:
+            logging.warning(f"import_entries() failed for status_info:{status_info}")
+            return None
+        else:
+            # STEP 3: Go back and fill out the rest of the "page" record based on all the resources.
+            agg_stats = ImportHarJson.aggregate_stats(
+                entries, page_id, first_url, first_html_url, status_info
+            )
+            if not agg_stats:
+                logging.warning(
+                    f"aggregate_stats() failed for status_info:{status_info}"
+                )
+                return None
+            else:
+                page.update(agg_stats)
+
+        return page, entries
+
+    # TODO finish porting logic
+    @staticmethod
+    def import_entries(entries, pageid, status_info):
+        requests = []
+        first_url = ""
+        first_html_url = ""
+
+        for entry in entries:
+            ret_request = {
+                "client": status_info["client"],
+                "date": status_info["date"],
+                "pageid": pageid,
+                "crawlid": status_info["crawlid"],
+                # we use this below for expAge calculation
+                "startedDateTime": utils.datetime_to_epoch(
+                    entry["startedDateTime"], status_info
+                ),
+                "time": entry["time"],
+                "_cdn_provider": entry.get("_cdn_provider"),
+                # amount response WOULD have been reduced if it had been gzipped
+                "_gzip_save": entry.get("_gzip_save"),
+            }
+
+            # REQUEST
+            try:
+                request = entry["request"]
+            except KeyError:
+                # TODO metric for failed parsing
+                logging.warning(
+                    f"Entry does not contain a request, status_info={status_info}, entry={entry}"
+                )
+                continue
+
+            url = request["url"]
+            (
+                request_headers,
+                request_other_headers,
+                request_cookie_size,
+            ) = utils.parse_header(
+                request["headers"], constants.ghReqHeaders, cookie_key="cookie"
+            )
+            ret_request.update(
+                {
+                    "method": request["method"],
+                    "httpVersion": request["httpVersion"],
+                    "url": url,
+                    "urlShort": url[:255],
+                    "reqHeadersSize": request.get("headersSize")
+                    if int(request.get("headersSize", 0)) > 0
+                    else None,
+                    "reqBodySize": request.get("bodySize")
+                    if int(request.get("bodySize", 0)) > 0
+                    else None,
```

Comment:
The formatting of these ternaries is a bit hard to read. Consider assigning them to variables outside of the dict, or using more indentation here.

Reply:
This formatting was enforced by using the `black` formatter requested by @tunetheweb in his review comments.

https://black.readthedocs.io/en/stable/

I have no preference, so will refactor them out with variable assignments.